File: DSA/src/go/remove_nth_node_from_end_of_list.go
package main

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	dummy := &ListNode{Val: 0, Next: head}
	first, second := dummy, dummy
	for n > 0 && second != nil {
		second = second.Next
		n--
	}
	for second != nil && second.Next != nil {
		first = first.Next
		second = second.Next
	}
	first.Next = first.Next.Next
	return dummy.Next
}

File: docs/DSA/src/py/remove_nth_node_from_end_of_list.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(val=0, next=head)
        first = second = dummy
        while second and n > 0:
            second = second.next
            n-=1

        while second.next:
            first = first.next
            second = second.next
        first.next = first.next.next
        return dummy.next
